Standards/scs-0216-v1-baseline-cluster-security.md

```diff
@@ -0,0 +1,141 @@
+| TCP      | 10252/10257 | kube-controller-manager | 10252 could be insecure before 1.13, after that only the secure port 10257 is available |
+| TCP      | 10255       | Read-Only Kubelet API   | HTTP port for GET requests                                                              |
+| TCP      | 30000-32767 | NodePort Services       | Service endpoints, could be HTTP                                                        |
+
+#### Authentication and Authorization
+
+All API clients should authenticate and authorize in order to be able to access an API or even
+specific functions of this API. This is the case for users as well as internal components.
+
+Most internal clients (like proxies or nodes) are typically authenticated via service accounts or
+x509 certificates, which will normally be created automatically during the setup of a cluster.
+External users can authenticate via an access pattern of choice, which is typically decided by
+the cluster provider.
+
+Authorization is (normally) done by he Role-Based Access Control (RBAC), which matches a request
+by a user with a set of permissions, also called a role. Kubernetes deploys some roles out-of-the-box;
+additional roles need to be carefully checked, since some permissions for specific resources allow
+modification of other resources.
+
+This whole process is especially important for the Kubelet, which allows anonymous requests in its
+default configuration. This is obviously a security risk, since everybody with access to its endpoint
+could manipulate resources that are managed with the Kubelet.
+
+To disable anonymous requests, the Kubelet should be started with `--anonymous-auth=false`.
+Authentication can be provided either through x509 client certificates or API bearer tokens.
+How to set up both approaches can be found in the [Kubelet Authentication and Authorization](https://kubernetes.io/docs/reference/access-authn-authz/kubelet-authn-authz/).
+
+Kubelet authorization is set to `AlwaysAllow` as a default mode. This can be quite problematic,
+since all authenticated users can do all actions. To mitigate this, it is possible to delegate
+authorization to the API server by:
+
+- enabling the `authorization.k8s.io/v1beta1` API group
+- starting the Kubelet with the `--authorization-mode=Webhook` and the `--kubeconfig` flags
+
+After that, the Kubelet calls the `SubjectAccessReview` API in order to determine the authorization of a request.
+
+## Decision
+
+This standard tries to increase security for a Kubernetes cluster and harden it in order
+to provide a high security setup. For this to work, multiple measures need to be undertaken.
+It is important to note that this standard is not REQUIRED for all clusters,
+but instead gives best practices for increasing security. Nevertheless, if a cluster is
+provided on the basis of high security, this standard MUST be applied.
```

Comment:
This paragraph still contains the "high security setup" term instead of "baseline security" or similar.

(The `Introduction` and `Motivation` sections also contain the "hardening" term, which might be a little bit overselling as well)

Reply:
Removed `hardening` and `high security` from the mentioned parts (and the rest of the document) and especially rewrote the commented section a bit more.

Reply:
Resolving this, since @joshmue gave his approval down below.